src/RA_CodeNotes.cpp

```diff
@@ -20,35 +22,39 @@ size_t CodeNotes::Load(const std::string& sFile)
     Clear();
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFile.c_str(), "rb") == 0)
+
+    std::ifstream ifile{ sFile, std::ios::binary };
+
+    Document doc;
+    IStreamWrapper isw{ ifile };
+    doc.ParseStream(isw);
+
+    if (!doc.HasParseError())
+
     {
-        Document doc;
-        doc.ParseStream(FileStream(pf));
-        if (!doc.HasParseError())
-        {
-            ASSERT(doc["CodeNotes"].IsArray());
+        ASSERT(doc["CodeNotes"].IsArray());
 
-            const Value& NoteArray = doc["CodeNotes"];
+        // use auto when you can to prevent implicit conversions (you get warnings on higher levels)
+        const auto& NoteArray = doc["CodeNotes"];
 
-            for (SizeType i = 0; i < NoteArray.Size(); ++i)
-            {
-                const Value& NextNote = NoteArray[i];
-                if (NextNote["Note"].IsNull())
-                    continue;
+        for (auto& i : NoteArray.GetArray())
+        {
+            if (i["Note"].IsNull())
```

Comment:
It might be more appropriate to use to an Expects/Ensures Contract to make sure it's not null. It's not in the standard yet but Microsoft has done a pretty good job at implemented Bjarne Stroustrup's C++ Core Guidelines.